packages/dashboard/src/pages/Personas/hooks/useContacts.ts

```diff
@@ -0,0 +1,47 @@
+import { useAsyncRetry, useUpdateEffect } from 'react-use'
+import { Services } from '../../../API'
+import type { Relation } from '@masknet/shared'
+import { useRef } from 'react'
+import { last } from 'lodash-es'
+
+export const useContacts = (network: string, page: number, size = 20) => {
+    const cache = useRef<Relation[]>([])
+
+    // If the network type be changed, clean cache
+    useUpdateEffect(() => {
+        if (cache.current.length) cache.current = []
+    }, [network])
```

Comment:
I suggest do not add a cache here. It's fragile and easy to get things wrong. We'll find a robust cache mechanism in the future.

Reply:
Yes, here is the error, I have replaced it with another way